cmd/mysterium_client/command_run/command.go

```diff
@@ -5,90 +5,94 @@ import (
 	MysteriumClient server.Client
 
-	DialogEstablisherFactory func(identity identity.Identity) communication.DialogEstablisher
-	dialog                   communication.Dialog
+	identityManager identity.IdentityManagerInterface
 
-	vpnMiddlewares []openvpn.ManagementMiddleware
-	vpnClient      *openvpn.Client
+	connectionManager client_connection.Manager
 
 	httpApiServer tequilapi.ApiServer
 }
 
-func (cmd *CommandRun) Run(options CommandOptions) (err error) {
-	consumerId := identity.FromAddress("consumer1")
+func NewCommand(options CommandOptions) (*CommandRun, error) {
+	nats_discovery.Bootstrap()
+	openvpn.Bootstrap()
 
-	session, err := cmd.MysteriumClient.SessionCreate(options.NodeKey)
-	if err != nil {
-		return err
-	}
-	proposal := session.ServiceProposal
-
-	dialogEstablisher := cmd.DialogEstablisherFactory(consumerId)
-	cmd.dialog, err = dialogEstablisher.CreateDialog(proposal.ProviderContacts[0])
-	if err != nil {
-		return err
-	}
+	mysteriumClient := server.NewClient()
 
-	vpnSession, err := vpn_session.RequestSessionCreate(cmd.dialog, proposal.Id)
-	if err != nil {
-		return err
-	}
+	keystoreInstance := keystore.NewKeyStore(options.DirectoryKeystore, keystore.StandardScryptN, keystore.StandardScryptP)
 
-	vpnConfig, err := openvpn.NewClientConfigFromString(
-		vpnSession.Config,
-		options.DirectoryRuntime+"/client.ovpn",
-	)
-	if err != nil {
-		return err
-	}
+	identityManager := identity.NewIdentityManager(keystoreInstance)
 
-	vpnMiddlewares := append(
-		cmd.vpnMiddlewares,
-		bytescount_client.NewMiddleware(cmd.MysteriumClient, session.Id, 1*time.Minute),
-	)
-	cmd.vpnClient = openvpn.NewClient(
-		vpnConfig,
-		options.DirectoryRuntime,
-		vpnMiddlewares...,
-	)
-	if err := cmd.vpnClient.Start(); err != nil {
-		return err
+	dialogEstablisherFactory := func(identity identity.Identity) communication.DialogEstablisher {
+		return nats_dialog.NewDialogEstablisher(identity)
 	}
 
-	router := tequilapi.NewApiRouter()
-
-	keystoreInstance := keystore.NewKeyStore(options.DirectoryKeystore, keystore.StandardScryptN, keystore.StandardScryptP)
-	idm := identity.NewIdentityManager(keystoreInstance)
-	endpoints.RegisterIdentitiesEndpoint(router, idm)
+	vpnClientFactory := configureVpnClientFactory(mysteriumClient, options.DirectoryRuntime)
 
-	endpoints.RegisterConnectionEndpoint(router, client_connection.NewManager())
+	vpnManager := client_connection.NewManager(mysteriumClient, dialogEstablisherFactory, vpnClientFactory)
 
-	cmd.httpApiServer, err = tequilapi.StartNewServer(options.TequilaApiAddress, options.TequilaApiPort, router)
+	httpApiServer, err := tequilapi.NewServer(options.TequilaApiAddress, options.TequilaApiPort)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	return nil
+	return &CommandRun{
+		mysteriumClient,
+		identityManager,
+		vpnManager,
+		httpApiServer,
+	}, nil
+}
+
+func (cmd *CommandRun) Run() {
+	router := tequilapi.NewApiRouter()
+	endpoints.RegisterIdentitiesEndpoint(router, cmd.identityManager)
```

Comment:
Should not this already be factories and injected?

Reply:
As per comment above - this can be moved to creation method